databallpy/features/covered_distance.py

```diff
@@ -0,0 +1,473 @@
+
+        dt = 1.0 / frame_rate
+
+        cols_to_drop = np.array(
+            [
+                [c + f"_{new_name}", c + f"_{new_name[0]}x", c + f"_{new_name[0]}y"]
+                for c in column_ids
+            ]
+        ).ravel()
+        df.drop(cols_to_drop, axis=1, errors="ignore", inplace=True)
+
+        res_dict = {}
+        for column_id in column_ids:
+            gradient_x = np.gradient(df[column_id + f"_{metric}x"].values, dt)
+            gradient_y = np.gradient(df[column_id + f"_{metric}y"].values, dt)
+            raw_differentiated = np.linalg.norm([gradient_x, gradient_y], axis=0)
+
+            # Scale gradients if magnitude exceeds max_val
+            if not pd.isnull(max_val):
+                exceed_max = raw_differentiated > max_val
+                scale_factor = max_val / raw_differentiated[exceed_max]
+                gradient_x[exceed_max] *= scale_factor
+                gradient_y[exceed_max] *= scale_factor
+
+            # smoothing the signal
+            if filter_type is not None:
+                gradient_x = _filter_data(
+                    gradient_x,
+                    filter_type=filter_type,
+                    window_length=window,
+                    polyorder=poly_order,
+                )
+                gradient_y = _filter_data(
+                    gradient_y,
+                    filter_type=filter_type,
+                    window_length=window,
+                    polyorder=poly_order,
+                )
+
+            res_dict[column_id + f"_{new_name[0]}x"] = np.array(gradient_x)
+            res_dict[column_id + f"_{new_name[0]}y"] = np.array(gradient_y)
+            res_dict[column_id + f"_{new_name}"] = np.linalg.norm(
+                [gradient_x, gradient_y], axis=0
+            )
+
+        new_columns_df = pd.DataFrame(res_dict)
+
+        if inplace:
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", category=pd.errors.PerformanceWarning)
+                df[new_columns_df.columns] = new_columns_df
+            return None
+        else:
+            return pd.concat([df, new_columns_df], axis=1)
+
+    except Exception as e:
+        LOGGER.exception(f"Found unexpected exception in _differentiate: \n{e}")
+        raise e
+
+
+#%% 3-7-2024
+def _parse_intervals(intervals):
+    if not all(isinstance(element, (int, float)) for element in intervals):
+        raise TypeError("All elements in the tuple must be integers or floats")
+    
+    if len(intervals) % 2 != 0:
+        raise ValueError("Intervals must contain an even number of elements.")
+    
+    return [(min(intervals[i], intervals[i+1]), max(intervals[i], intervals[i+1])) for i in range(0, len(intervals), 2)]
+
+def get_covered_distance(
+    tracking_data: pd.DataFrame,
+    player_ids: list[str],
+    framerate: int,
+    velocity_intervals: tuple[float, ...] | None = None,
+    acceleration_intervals: tuple[float, ...] | None = None
```

Comment:
What did we decide on a start/end idx of the data frame?

Maybe add something like
`start_idx = tracking_data.index[0] if start_idx is None else start_idx`
`end_idx = tracking_data.index[-1] if end_idx is None else end_idx`
tracking_data = tracking_data.loc[start_idx:end_idx]